core/include/thread.h

```diff
@@ -209,6 +209,9 @@ struct _thread {
     const char *name;               /**< thread's name                  */
     int stack_size;                 /**< thread's stack size            */
 #endif
+#ifdef HAVE_THREAD_ARCH_STATE_T
```

Comment:
may sound nitpicky, but do you insist on the "state" part of the variable? it is not part of ```thread_t```, and as this is a data type, implicitly it contains "(state) data", so IMO it can be shortened.
What do you think?

Reply:
Sorry, I was not clear. I was just suggesting renaming ```thread_arch_state_t``` to ```thread_arch_t```. The feature is very much wanted!

Sorry again!

Reply:
Changed and rebased. Description updated.